modal/_asgi.py

```diff
@@ -71,10 +86,10 @@ async def send(msg):
                     indices = list(range(0, size, chunk_size))
                     for i in indices[:-1]:
                         chunk = msg["body"][i : i + chunk_size]
-                        await messages_from_app.put({"type": "http.response.body", "body": chunk, "more_body": True})
+                        await messages_from_app.put([{"type": "http.response.body", "body": chunk, "more_body": True}])
```

Comment:
Why did we turn this into lists?

Reply:
To ensure atomicity when adding multiple "related" messages at once (which we do in when the input generation task gets an error). At some point I managed to get into a weird error state where our own internal error got intermixed with responses from the app.

Reply:
Hmm, we're still `yield`-ing individual items though, so there's no guarantee that they'll be put on the `DataChannel` at the same time. Also I don't think our internal error is sent on this asyncio Queue, since the exception goes directly through the output channel

Reply:
No, I mean the internal error we produce in the timeout of first-input, which still uses the `messages_from_app` Queue and added multiple messages there sequentially before this patch:

```
await messages_from_app.put({"type": "http.response.start", "status": 502})
await messages_from_app.put(
    {
    "type": "http.response.body",
     "body": b"Missing request, possibly due to cancellation or crash",
}
```

We should probably also ensure in the http case that these messages only get sent/queued if the asgi app hasn't already started sending responses and vice versa (but maybe modal-http would error if multiple `http.response.start` came in?)